crates/pop-drink/src/lib.rs

```diff
@@ -1,95 +1,55 @@
-        let account: [u8; 32] = s.clone().into();
-        utils::account_id_from_slice(&account)
-    }
-}
+	/// A set of primitive runtime errors and versioned runtime errors used for testing.
+	pub mod error {
+		pub use pop_runtime_devnet::RuntimeError::*;
+
+		pub use crate::error::*;
+
+		/// A collection of error types from the `v0` module used for smart contract testing in the
+		/// `devnet` environment.
+		pub mod v0 {
+			use super::*;
+			pub use crate::error::v0::*;
+
+			/// Configuration for the runtime error used to test smart contract.
+			#[derive(Debug)]
+			pub struct Config;
+			impl RuntimeErrorConfig for Config {
+				type ModuleError = pop_runtime_devnet::RuntimeError;
+				type PopApiError = pop_primitives::v0::Error;
 
-pub mod utils {
-    use super::*;
-    pub fn deploy<S, E>
-    (
-        session: &mut Session<S>,
-        bundle: ContractBundle,
-        method: &str,
-        input: Vec<String>,
-        salt: Vec<u8>,
-        init_value: Option<BalanceFor<S::Runtime>>,
-    )
-        -> Result<AccountIdFor<S::Runtime>, E>
-        where
-            S: Sandbox,
-            S::Runtime: pallet_contracts::Config,
-            E: Decode,
-    {
-        let result = session.deploy_bundle(bundle, method, &input, salt, init_value);
-        if result.is_err() {
-            let deployment_result = session.record().last_deploy_result().result.clone();
-            let error = deployment_result.unwrap().result.data;
-            return Err(E::decode(&mut &error[2..]).unwrap());
-        }
-        Ok(result.unwrap())
-    }
+				const MODULE_INDEX: u8 = 3;
+			}
 
-    /// Call a contract method and decode the returned data.
-    pub fn call<S, O, E>(
-        session: &mut Session<S>,
-        func_name: &str,
-        input: Vec<String>,
-        endowment: Option<BalanceFor<S::Runtime>>,
-    ) -> Result<O, E>
-    where
-        S: Sandbox,
-        S::Runtime: pallet_contracts::Config,
-        O: Decode,
-        E: Decode,
-    {
-        match session.call::<String, ()>(func_name, &input, endowment) {
-            // If the call is reverted, decode the error into `PSP22Error`.
-            Err(SessionError::CallReverted(error)) =>
-                Err(E::decode(&mut &error[2..])
-                    .unwrap_or_else(|_| panic!("Decoding failed"))),
-            // If the call is successful, decode the last returned value.
-            Ok(_) => Ok(session
-                .record()
-                .last_call_return_decoded::<O>()
-                .unwrap_or_else(|_| panic!("Expected a return value"))
-                .unwrap_or_else(|_| panic!("Decoding failed"))),
-            // Catch-all for unexpected results.
-            _ => panic!("Expected call to revert or return a value"),
-        }
-    }
+			/// Runtime error for testing contracts using the Pop API V0.
+			pub type RuntimeError = crate::error::RuntimeError<Config>;
```

Comment:
```suggestion
			pub type Error = crate::error::Error<runtime::RuntimeError, pop_primitives::v0::Error, 3>;
```

Using just 2 extra generic parameters eliminates a lot of boilerplate code.

Reply:
Resolved